doc/influxql-2.md

```diff
@@ -0,0 +1,910 @@
+
+select(database:"foo",where:{'metric' = "cpu"})
+  .range(start:-1h)
+  .window(period:10m,start:-1h,round:10m)
+// if executed at 13:03:59, windows would occur at 12:00:00, 12:10:0, 12:20:0, etc.
+```
+
+#### Merge
+`merge` will merge series in a matrix together based on the grouping criteria. Note that all values in a series must be of the same type. If a type difference is encountered, merge will cast based on the type of the first series in the matrix. You can use the type casting functions to force a consistent type before a call to merge. (should we have merge throw an error instead and direct the user to typecast?)
+
+```javascript
+// Accepts: matrix
+// Arguments:
+//   keys: an array of strings. This acts like a group by clause in SQL
+//   keep: an array of strings. These are tag key data that should be kept in the individual points
+//   error: boolean to return an error on type casting issue. Defaults to false
+// Returns: matrix
+select(database:"foo",where:{'metric'="network_in"})
+  .merge(keys:["region"])
+```
+
+We'll use tables as an example to show how it works. For example, if we have the following matrix `foo`:
+
+| Series                                      |  10 |  20 |
+|---------------------------------------------|:---:|:---:|
+|`{"host":"a","metric":"network","dc":"west"}`|  1  |  1  |
+|`{"host":"b","metric":"network","dc":"east"}`|  2  |  2  |
+|`{"host":"c","metric":"network","dc":"east"}`|  3  |  3  |
+|`{"host":"d","metric":"network","dc":"west"}`|  4  |  4  |
+
+If we called `foo.merge(keys:["metric"])` we would get a matrix that looks something like this:
+
+| Series               |10 `host:a,dc:west`|10 `host:b,dc:east`|10 `host:c,dc:east`|10 `host:d,dc:west`|20 `host:a,dc:west`|20 `host:b,dc:east`|20 `host:c,dc:east`|20 `host:d,dc:west`|
+|----------------------|:---:|:---:|:---:|:---:|:---:|:---:|:---:|:---:|
+|`{"metric":"network"}`|  1  |  2  |  3  |  4  |  1  |  2  |  3  |  4  |
+
+Note that in results that get returned to the client, only the series information would be retained, unless we pass in an argument to keep certain keys as tagsets on the individual points.
+
+We can merge based on multiple tag keys. For example if we called `foo.merge(keys:["metric", "dc"]) we would get a matrix that looks like this:
+
+| Series                           |  10 `host:a`| 10 `host:b`| 10 `host:c`| 10 `host:d`| 20 `host:a`| 20 `host:b`| 20 `host:c`| 20 `host:d`|
+|----------------------------------|:---:|:---:|:---:|:---:|:---:|:---:|:---:|:---:|
+|`{"metric":"network","dc":"west"}`|  1  |     |     |  4  |  1  |     |     |  4  |
+|`{"metric":"network","dc":"east"}`|     |  2  |  3  |     |     |  2  |  3  |     |
+
+#### Keys
+`keys` will return an array of tag keys that occur in the matrix. The output of keys function can be used as the predecessor argument to `empty` or `not empty` in a `where` expression. For pagination through keys, `limit` can be chained off it.
+
```

Comment:
For that you'd have to have a `range`, but it might not have the best performance. For instance: `select(database:"foo").range(start:-4h).last().keys()`